pkg/deploy/apply.go

```diff
@@ -136,7 +136,10 @@ func withAwaitableResource(apply applyFunction) applyFunction {
 		// consume watcher events and wait for the resource to complete or exit because of timeout
 		for {
 			select {
-			case event := <-watchEvents:
+			case event, ok := <-watchEvents:
```

Comment:
Can you add some test to replicate the bug?

Reply:
Ok, i added some tests in particular [this one](https://github.com/mia-platform/mlp/pull/60/files#diff-95f792437e0ecf57348a18366d9c98b61904f3276694b2335ee30ad075f9ab37R610) that replicates the bug

Reply:
We had a precision problem comparing the start time (in nanoseconds) and event time (in seconds)